plugins/inventory/vmware_vm_inventory.py

```diff
@@ -68,19 +69,69 @@
             description:
             - Specify the list of VMware schema properties associated with the VM.
             - These properties will be populated in hostvars of the given VM.
-            - Each value in the list specifies the path to a specific property in VM object.
+            - Each value in the list can be a path to a specific property in VM object or a path to a collection of VM objects.
+            - In addition to VM properties, the following are special values
+            - Use 'customValue' to populate VM's custom attributes.
+            - Use 'all' to populate all VM properties.
             type: list
             default: [ 'name', 'config.cpuHotAddEnabled', 'config.cpuHotRemoveEnabled',
                        'config.instanceUuid', 'config.hardware.numCPU', 'config.template',
-                       'config.name', 'guest.hostName', 'guest.ipAddress',
+                       'config.name', 'config.uuid' ,'guest.hostName', 'guest.ipAddress',
                        'guest.guestId', 'guest.guestState', 'runtime.maxMemoryUsage',
-                       'customValue'
+                       'customValue', 'summary.runtime.powerState', 'config.guestId'
                        ]
+            version_added: "2.9"
+        resources:
+            description:
+            - A list of resources to limit search scope.
+            - Each resource item is represented by exactly one C('vim_type_snake_case):C(list of resource names) pair and optional nested I(resources)
+            - Key name is based on snake case of a vim type name; e.g C(host_system) correspond to C(vim.HostSystem)
+            - See  L(VIM Types,https://pubs.vmware.com/vi-sdk/visdk250/ReferenceGuide/index-mo_types.html)
+            version_added: "2.10"
+            required: False
+            type: list
```

Comment:
does this let you declare the type of elements, like another entry here `elements: dict`? they are enforcing this for the modules.  This looks very similar to the host filters, but looks like it doesn't quite do it?